addons/sale_timesheet/models/sale_order.py

```diff
@@ -169,6 +169,19 @@ def _compute_product_updatable(self):
             else:
                 super(SaleOrderLine, line)._compute_product_updatable()
 
+    def _init_column(self, column_name):
+        if column_name != "is_service":
+            return super()._init_column(column_name)
+        self.env.cr.execute("""
+            UPDATE sale_order_line line
+            SET is_service = (pt.type = 'service')
+            FROM product_product pp
+            LEFT JOIN product_template pt ON pt.id = pp.product_tmpl_id
+            WHERE pp.id = line.product_id
+        """)
+        # Skip initial computation via _compute_is_service
+        return True
```

Comment:
Document this feature in the docstring of the original `_init_column` column and make the original method return `False`.